tests/components/comelit/test_alarm_control_panel.py

```diff
@@ -0,0 +1,155 @@
+"""Tests for Comelit SimpleHome alarm control panel platform."""
+
+from unittest.mock import AsyncMock
+
+from aiocomelit.api import AlarmDataObject, ComelitVedoAreaObject, ComelitVedoZoneObject
+from aiocomelit.const import AlarmAreaState, AlarmZoneState
+from freezegun.api import FrozenDateTimeFactory
+import pytest
+
+from homeassistant.components.alarm_control_panel import (
+    ATTR_CODE,
+    DOMAIN as ALARM_DOMAIN,
+    SERVICE_ALARM_ARM_AWAY,
+    SERVICE_ALARM_ARM_HOME,
+    SERVICE_ALARM_ARM_NIGHT,
+    SERVICE_ALARM_DISARM,
+    AlarmControlPanelState,
+)
+from homeassistant.components.comelit.const import SCAN_INTERVAL
+from homeassistant.const import ATTR_ENTITY_ID, STATE_UNAVAILABLE
+from homeassistant.core import HomeAssistant
+
+from . import setup_integration
+from .const import VEDO_PIN
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+ENTITY_ID = "alarm_control_panel.area0"
+
+
+@pytest.mark.parametrize(
+    ("human_status", "armed", "alarm_state"),
+    [
+        (AlarmAreaState.DISARMED, 0, AlarmControlPanelState.DISARMED),
+        (AlarmAreaState.ARMED, 1, AlarmControlPanelState.ARMED_HOME),
+        (AlarmAreaState.ARMED, 2, AlarmControlPanelState.ARMED_HOME),
+        (AlarmAreaState.ARMED, 3, AlarmControlPanelState.ARMED_NIGHT),
+        (AlarmAreaState.ARMED, 4, AlarmControlPanelState.ARMED_AWAY),
+        (AlarmAreaState.UNKNOWN, 0, STATE_UNAVAILABLE),
+    ],
+)
+async def test_entity_availability(
+    hass: HomeAssistant,
+    freezer: FrozenDateTimeFactory,
+    mock_vedo: AsyncMock,
+    mock_vedo_config_entry: MockConfigEntry,
+    human_status: AlarmAreaState,
+    armed: int,
+    alarm_state: AlarmControlPanelState,
+) -> None:
+    """Test all entities."""
+
+    await setup_integration(hass, mock_vedo_config_entry)
+
+    assert (state := hass.states.get(ENTITY_ID))
+    assert state.state == AlarmControlPanelState.DISARMED
+
+    vedo_query = AlarmDataObject(
+        alarm_areas={
+            0: ComelitVedoAreaObject(
+                index=0,
+                name="Area0",
+                p1=True,
+                p2=True,
+                ready=False,
+                armed=armed,
+                alarm=False,
+                alarm_memory=False,
+                sabotage=False,
+                anomaly=False,
+                in_time=False,
+                out_time=False,
+                human_status=human_status,
+            )
+        },
+        alarm_zones={
+            0: ComelitVedoZoneObject(
+                index=0,
+                name="Zone0",
+                status_api="0x000",
+                status=0,
+                human_status=AlarmZoneState.REST,
+            )
+        },
+    )
+
+    mock_vedo.get_all_areas_and_zones.return_value = vedo_query
+
+    freezer.tick(SCAN_INTERVAL)
+    async_fire_time_changed(hass)
+    await hass.async_block_till_done(wait_background_tasks=True)
```

Comment:
I didn't find any background tasks for in the integration, is it really needed?

Reply:
Yes, the coordinator polling needs to be aware that time passed: https://github.com/home-assistant/core/blob/dev/homeassistant/components/comelit/coordinator.py#L56

Reply:
Coordinator is aware of the time passed, waiting for background tasks is needed only if you specifically started background tasks.
Do you have any background tasks running?
Here is the same example for Shelly:
https://github.com/home-assistant/core/blob/aa2ab74ee98bb4c479f4d46c79d2d0312bdcda94/tests/components/shelly/test_sensor.py#L1445-L1447

Reply:
Code updated